### PR TITLE
[FLINK-30255] Throw exception when names are upper case in HiveCatalog

### PR DESCRIPTION
**Pull request by zjureel**

### CODE REVIEW

**Comment by zjureel** [comment]
Hi @JingsongLi Can you help to review this PR? THX

File: flink-table-store-hive/flink-table-store-hive-catalog/src/main/java/org/apache/flink/table/store/hive/HiveCatalog.java [hunk 1]
```diff
@@ -269,6 +271,26 @@ protected String warehouse() {
...
+                        objectPath.getDatabaseName()));
+        checkState(
+                objectPath.getObjectName().equals(objectPath.getObjectName().toLowerCase()),
+                String.format(
+                        "Table name[%s] cannot contain upper case", objectPath.getObjectName()));
+    }
+
+    private void checkFieldNamesUpperCase(List<String> fieldNames) {
+        for (String fieldName : fieldNames) {
+            checkState(
```
Comment (SteNicholas) [inline flink-table-store-hive/flink-table-store-hive-catalog/src/main/java/org/apache/flink/table/store/hive/HiveCatalog.java:288]: Could we check the state of `fieldNames` not single `fieldName`? IMO, it's better to provide users with all upper case field names once.

File: flink-table-store-hive/flink-table-store-hive-catalog/src/test/java/org/apache/flink/table/store/hive/HiveCatalogITCase.java [hunk 2]
```diff
@@ -317,6 +318,25 @@ public void testHiveLock() throws InterruptedException {
...
+    public void testUpperCase() {
+        assertThatThrownBy(
+                        () ->
+                                tEnv.executeSql(
+                                                "CREATE TABLE T ( a INT, b STRING ) WITH ( 'file.format' = 'avro' )")
+                                        .await())
+                .hasRootCauseMessage(
+                        String.format("Table name[%s] cannot contain upper case", "T"));
+
+        assertThatThrownBy(
```
Comment (SteNicholas) [inline flink-table-store-hive/flink-table-store-hive-catalog/src/test/java/org/apache/flink/table/store/hive/HiveCatalogITCase.java:331]: Add the test cases for multiple upper case field names.

**Comment by zjureel** [comment]
Thanks @SteNicholas I have updated the codes

**Review by JingsongLi — APPROVED** [review]
Looks good to me!

**Pull request merged**